week2/Program.cs

```diff
@@ -0,0 +1,183 @@
+    for (int i = 0; i <= 100; i++)
+    {
+        numbers.Add(i);
+    }
+    return numbers;
+};
+
+app.MapGet("/prep-task6", () =>
+{
+    List<int> numbersToHundredList = NumbersToHundredList();
+    numbersToHundredList.RemoveRange(33, 12);
+    //numbersToHundredList.ForEach(Console.WriteLine);
+    return numbersToHundredList;
+});
+
+app.MapGet("/prep-task7", () =>
+{
+    // I did only 1st part of the 7th exercise - to split the list.
+    // But couldn't proceed to next as the return type here is in string.
+    // I tried Chunk and GroupBY to split the list and its return is in string.
+    List<int> numbersToHundredList = NumbersToHundredList();
+    int numberOfElementsInSplittedList = (numbersToHundredList.Count + 1) / 2;
+
+    string splittedList = System.Text.Json.JsonSerializer.Serialize(numbersToHundredList.Chunk(numberOfElementsInSplittedList));
+    return splittedList;
+
+    // var splittedList = numbersToHundredList.Select((x, idx) => new { index = idx, val = x })
+    //         .GroupBy(i => i.index / numberOfElementsInSplittedList)
+    //         .Select(g => g.Select(x => x.val));
+    // return splittedList;
+});
+
+// Class exercises
+app.MapGet("/class-task1", (string number1, string number2, string operation) =>
+{
+    bool numberOneIsValid = int.TryParse(number1, out int num1);
+    bool numberTwoIsValid = int.TryParse(number2, out int num2);
+    int result = 0;
+    if (!numberOneIsValid && !numberTwoIsValid)
+    {
+        return Results.BadRequest("Not a valid number");
+    }
+    switch (operation)
+    {
+        case "add":
+            result = num1 + num2;
+            break;
+        case "substract":
+            result = num1 - num2;
+            break;
+        case "multiply":
+            result = num1 * num2;
+            break;
+        default:
+            return Results.BadRequest("Not a valid operation");
+    }
+    return Results.Ok(result);
+
+});
+
+app.MapGet("/class-task2", (string input) =>
+{
+    bool isNumber = int.TryParse(input, out var number);
+    if (isNumber) return Results.Ok(AddNumbers(number));
+    return Results.Ok(CountCapitalLetters(input));
+
+    int AddNumbers(int input)
+    {
+        int[] digits = input.ToString().Select(digit => Convert.ToInt32(digit.ToString())).ToArray();
+        //var digits = input.ToString().Select(digit => int.Parse(digit.ToString())).ToArray();
+        int results = 0;
+        foreach (int i in digits)
+        {
+            results += i;
+        }
+        return results;
+    };
+
+    int CountCapitalLetters(string input)
+    {
+        int count = 0;
+        foreach (char c in input)
+        {
+            if (char.IsUpper(c))
+                count++;
+        }
+        return count;
+    }
+});
+
+app.MapGet("/class-task3", (string input) =>
+{
+    var result = new List<char>();
+    char[] characters = input.ToLower().ToCharArray();
+    foreach (char c in characters)
+    {
+        if (!Char.IsWhiteSpace(c))

```

Comment:
Really cool, a better approach to detecting alphanumeric characters is to use `Char.IsLetterOrDigit`.

You also need to check if a character is not already part of the list. You can do it by:
```suggestion
        if (!Char.IsWhiteSpace(c) && !characters.Contains(input[i]))
```

And another way, that we don't expect students to do, is by using [distinct](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.distinct?view=net-7.0) & `OrderBy` extension method. Here we could apply expression body since it would be a one-liner.
```
app.MapGet("/alphabetical", (string input) => input.Distinct().ToList().OrderBy(s => s);
```

Reply:
Thanks Ivan for the great feedback 👍